src/plugins/vis_type_pie/public/utils/get_layers.ts

```diff
@@ -30,14 +30,30 @@ export const computeColor = (
   rows: DatatableRow[],
   visParams: PieVisParams,
   palettes: PaletteRegistry | null,
-  syncColors: boolean
+  syncColors: boolean,
+  formatter: DataPublicPluginStart['fieldFormats'],
+  format?: BucketColumns['format']
 ) => {
   const { parentSeries, allSeries } = getDistinctSeries(rows, columns);
+  const dataName = d.dataName;
+
+  let formattedLabel = '';
+  if (format) {
+    formattedLabel = formatter.deserialize(format).convert(dataName) ?? '';
+  }
 
   if (visParams.distinctColors) {
-    const dataName = d.dataName;
+    let overwriteColor;
+    if (Object.keys(overwriteColors).includes(formattedLabel)) {
+      overwriteColor = overwriteColors[formattedLabel];
```

Comment:
If I'm understanding correctly this is only here for legacy reasons and will never be generated by the new pie implementation, right? If yes, could we leave a comment here explaining why this code branch is needed? Otherwise it might be removed at some point.

Reply:
Yes Joe, you are right. I will add a comment